client/rest-high-level/src/test/java/org/elasticsearch/client/ESRestHighLevelClientTestCase.java

```diff
@@ -70,6 +70,17 @@ protected static RestHighLevelClient highLevelClient() {
         }
     }
 
+    protected static <Req, Resp> Resp executeWithOptions(Req request, SyncMethodWithRequestOptions<Req, Resp> syncMethod,
+                                              AsyncMethodWithRequestOptions<Req, Resp> asyncMethod, RequestOptions options) throws IOException {
```

Comment:
after some discussion with @nik9000, I tried refactoring the existing `execute` and associated functional interfaces, and it seems like a big enough refactor that I would rather not include in this PR. I'm happy to follow up on a separate PR, but these changes make it possible to migrate the `*IT.java` tests piecemeal.

Reply:
gotcha, this has to do with #31069 then, which I will get in today hopefully, then no refactorings should be needed.